transport/http/config.go

```diff
@@ -130,16 +130,26 @@ func (ts *transportSpec) buildTransport(tc *TransportConfig, k *yarpcconfig.Kit)
 //  inbounds:
 //    http:
 //      address: ":80"
+//      grabHeaders:
```

Comment:
This isn't something I imagine having to configure * every service - instead, wouldn't we want to see this done once in a central place?

Reply:
How would you suggest that?

Reply:
I suppose I imagined a transport option that would be passed in centrally, like in yarpcfx for example. With this approach, the guidance would essentially lead to a whole bunch of copy/pasting config between services, no?

Reply:
Meaning I think the `GrabHeaders` option is perfect, but I'm not sure we should expose it in config just yet. Let's ask Mike what he envisioned.

Reply:
I don't think it is any harm to be able to config it, it's not like this will need to be deprecated as long as we have the `GrabHeaders` option, right?

Reply:
More that we can't remove it once we add it, so I'd rather be conservative unless we know we need to expose it. 

Remember that everything we add to config needs to be documented in end-user guides and communicated.

Reply:
OK so should I just remove it? Or do we want to wait?

Reply:
Note that by my recollection, every other option in the transports other than settings for the Tracer have a corresponding config setting. If we didn't have one here, this would effectively be special-cased I think. This is just following the pattern.

Reply:
```go
// TransportConfig configures the shared HTTP Transport. This is shared
// between all HTTP outbounds and inbounds of a Dispatcher.
//
//  transports:
//    http:
//      keepAlive: 30s
//      maxIdleConnsPerHost: 2
//      connTimeout: 500ms
//      connBackoff:
//        exponential:
//          first: 10ms
//          max: 30s
//
// All parameters of TransportConfig are optional. This section may be omitted
// in the transports section.
type TransportConfig struct {
	// Specifies the keep-alive period for all HTTP clients. This field is
	// optional.
	KeepAlive           time.Duration       `config:"keepAlive"`
	MaxIdleConnsPerHost int                 `config:"maxIdleConnsPerHost"`
	ConnTimeout         time.Duration       `config:"connTimeout"`
	ConnBackoff         yarpcconfig.Backoff `config:"connBackoff"`
}
```

Reply:
I'm not going to blocked.

I was hoping to convince you that we can always add it later, and shouldn't add things unless there is a specific ask because we'll be committing to them for all of 1.x and you'll have to circle back and write end-user guides and communicate this.